### PR TITLE
Autocomplete functionality

### LINKED ISSUE #114: Ajax-y autocomplete location options
Populate destination search

### PR DESCRIPTION
**Pull request by njgheorghita**
Turns out we don't really need ajax thanks to the autocomplete widget in jquery-ui to complete this functionality, though I'll ask Josh tomorrow about the performance between this strategy and straight up ajax.

closes #114

### CODE REVIEW

File: app/models/couch.rb [hunk 1]
```diff
@@ -16,6 +16,10 @@ def self.search(params)
       .distinct
   end
 
+  def self.availableCities
+    Couch.all.pluck(:city).uniq
```
Comment (tmikeschu) [inline app/models/couch.rb]: Totally nit picking, but do we need to camel case this Ruby method?

File: app/controllers/home_controller.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 class HomeController < ApplicationController
   def show
+    gon.availableCities = Couch.availableCities
```
Comment (tmikeschu) [inline app/controllers/home_controller.rb]: I love how this works. Do you have any idea about how JS can read this without making it an ivar?

**Review by tmikeschu — COMMENTED** [review]
Nice work! All about finding the simplest solution.

Just one comment about camel casing ruby vars, but I'm all thumbs up for this.

File: app/models/couch.rb (hunk 1 above)
Comment (ski-climb) [reply app/models/couch.rb]: second Schutte's camelCase thoughts.

**Review by ski-climb — COMMENTED** [review]
Other than the camel case issue that Schutte mentioned, it looks good.  Nice find!

**Comment by riverswb** [comment]
Looks good, other than camel case and conflict

**Comment by njgheorghita** [comment]
@tmikeschu @riverswb @ski-climb @Sh1pley Converted camel to snake. @tmikeschu I see what you're saying about the instance variable, I'll keep on playing with it to see if I can find a better solution while I mess around with AJAX. This pr should be good to merge, if one of y'all would like to do the honors.

**Pull request merged**